From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 30 Aug 2018 07:11:24 +0200
Subject: search: Ignore search provider results metas if search is cancelled

Call updateSearch callback with no results when the search provider has been
cancelled, without doing any logging.

Bug-GNOME: https://gitlab.gnome.org/GNOME/gnome-shell/issues/183
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/bionic/+source/gnome-shell/+bug/1756826
Forwarded: https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/205
---
 js/ui/search.js | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/js/ui/search.js b/js/ui/search.js
index 804be95..dd4bfad 100644
--- a/js/ui/search.js
+++ b/js/ui/search.js
@@ -227,8 +227,9 @@ var SearchResultsBase = new Lang.Class({
 
             this.provider.getResultMetas(metasNeeded, metas => {
                 if (metas.length != metasNeeded.length) {
-                    log('Wrong number of result metas returned by search provider ' + this.provider.id +
-                        ': expected ' + metasNeeded.length + ' but got ' + metas.length);
+                    if (!this._cancellable.is_cancelled())
+                        log(`Wrong number of result metas returned by search provider ${this.provider.id}` +
+                            `: expected ${metasNeeded.length} but got ${metas.length}`);
                     callback(false);
                     return;
                 }
